ttt-lexical-patterns.lisp

```diff
@@ -9,6 +9,29 @@
 (defparameter *tense* '(past pres cf))
 (defparameter *coordinator* '(and or but because))
 (defparameter *detformer* '(nquan fquan))
+(defparameter *semtypes*
+  '(("([A-Z]|_)*\\.PRO" . "D")
```

Comment:
Use `\\[?\\{?\(\\w\|\\d\|-\|\/\|:\|\\.\|\\*\|\\[\|\\]\)\+\\}?` instead of `([A-Z]|_)*`.  ULF syntactically allows a bunch of symbols and there are special cases like implicit atoms (e.g. `{ref}.pro`). There should probably be a variable for this from which the other patterns are generated so that we don't hard code this pattern in multiple places. (BTW, this pattern was developed over many iterations of my past work and it the one that is used in ttt-lexical-patterns for the current hacky type analysis; there was no way I would have expected you to get this..)

Also, add `^` and `$` to the start and end of the patterns, respectively so that we don't allow substring matches. (e.g. "^([A-Z]|_)*\\.PRO$")

Reply:
Substring matches aren't a problem because of the way my lambda is matching the patterns (in my `assoc` call). I'm matching the string with the pattern and then comparing the string returned back with the original string.

Reply:
Alright, that's fine then.